include/fiction/algorithms/simulation/sidb/determine_groundstate_from_simulation_results.hpp

```diff
@@ -0,0 +1,77 @@
+//
+// Created by Jan Drewniok on 12.02.24.
+//
+
+#ifndef FICTION_DETERMINE_GROUNDSTATE_FROM_SIMULATION_RESULTS_HPP
+#define FICTION_DETERMINE_GROUNDSTATE_FROM_SIMULATION_RESULTS_HPP
+
+#include "fiction/algorithms/simulation/sidb/sidb_simulation_result.hpp"
+#include "fiction/technology/charge_distribution_surface.hpp"
+
+#include <algorithm>
+#include <cstdint>
+#include <limits>
+#include <set>
+#include <vector>
+
+namespace fiction
+{
+
+/**
+ * This function calculates the ground state charge distributions from the provided simulation results.
+ * The ground state charge distributions are those with energy closest to the minimum energy found in the simulation
+ * results.
+ *
+ * @note When degenerate states exist, there are multiple ground states with the same energy.
+ *
+ * @tparam Lyt The layout type used in the simulation results.
+ * @param simulation_results The simulation results containing charge distributions.
+ * @return A vector of charge distributions with the minimal energy.
+ */
+template <typename Lyt>
+[[nodiscard]] std::vector<charge_distribution_surface<Lyt>>
+determine_groundstate_from_simulation_results(const sidb_simulation_result<Lyt>& simulation_results) noexcept
+{
+    std::vector<charge_distribution_surface<Lyt>> groundstate_charge_distributions{};
+    std::set<uint64_t>                            charge_indices{};
+
+    // in case, the charge indices are not updated.
+    auto charge_configurations_copy = simulation_results.charge_distributions;
+
+    // Find all unique charge indices. This is done because simulation results can have multiple identical charge
+    // distributions.
+    for (auto& cds : charge_configurations_copy)
```

Comment:
Not sure, but can this be `const`? The functions you call on `cds` sound like they could be all `const`.